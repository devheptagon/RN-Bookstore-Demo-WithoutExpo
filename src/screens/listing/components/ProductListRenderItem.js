import React from 'react'
import PropTypes from 'prop-types'
import { TouchableHighlight } from 'react-native'
import ProductListItem from './ProductListItem'
import { ProductListingTypes } from '../../../constants/Enums'
import BasketListItem from './BasketListItem'

export const ProductListRenderItem = props => {
    const { onPress, item, index, listingType } = props
    switch (listingType) {
        case ProductListingTypes.SearchResults:
            return (
                <TouchableHighlight onPress={onPress}>
                    <ProductListItem dataItem={item} rowIndex={index} />
                </TouchableHighlight>
            )
        case ProductListingTypes.Basket:
            return <BasketListItem dataItem={item} rowIndex={index} />
        default:
            return null
    }
}

ProductListRenderItem.propTypes = {  
    onPress: PropTypes.func, 
    item: PropTypes.object, 
    index: PropTypes.number, 
    listingType: PropTypes.number
}

export default ProductListRenderItem