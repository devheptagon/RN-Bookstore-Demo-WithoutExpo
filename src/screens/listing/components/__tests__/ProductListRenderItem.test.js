import React from 'react';
import {ProductListRenderItem} from '../ProductListRenderItem';
import renderer from 'react-test-renderer';

test('ProductListRenderItem renders correctly', () => {
  const tree = renderer.create(<ProductListRenderItem />).toJSON();
  expect(tree).toMatchSnapshot();
});