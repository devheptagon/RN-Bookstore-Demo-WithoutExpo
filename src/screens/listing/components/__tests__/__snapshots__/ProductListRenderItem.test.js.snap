// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListRenderItem renders correctly 1`] = `null`;
